modules/apps/27-interchain-accounts/keeper/relay_test.go

```diff
@@ -91,3 +100,128 @@ func (suite *KeeperTestSuite) TestTrySendTx() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestOnRecvPacket() {
+	var (
+		path       *ibctesting.Path
+		msg        sdk.Msg
+		txBytes    []byte
+		packetData []byte
+		sourcePort string
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"Interchain account successfully executes banktypes.MsgSend", func() {
+				// build MsgSend
+				amount, _ := sdk.ParseCoinsNormalized("100stake")
+				interchainAccountAddr, _ := suite.chainB.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), path.EndpointA.ChannelConfig.PortID)
+				msg = &banktypes.MsgSend{FromAddress: interchainAccountAddr, ToAddress: suite.chainB.SenderAccount.GetAddress().String(), Amount: amount}
+				// build packet data
+				txBytes, err := suite.chainA.GetSimApp().ICAKeeper.SerializeCosmosTx(suite.chainA.Codec, msg)
+				suite.Require().NoError(err)
+
+				data := types.IBCAccountPacketData{Type: types.EXECUTE_TX,
+					Data: txBytes}
+				packetData = data.GetBytes()
+			}, true,
+		},
+		{
+			"Cannot deserialize txBytes", func() {
+				txBytes = []byte("invalid tx bytes")
+				data := types.IBCAccountPacketData{Type: types.EXECUTE_TX,
+					Data: txBytes}
+				packetData = data.GetBytes()
+			}, false,
+		},
+		{
+			"Cannot deserialize txBytes: invalid IBCTxRaw", func() {
+				txBody := []byte("invalid tx body")
+				txRaw := &types.IBCTxRaw{
+					BodyBytes: txBody,
+				}
+
+				txBytes = suite.chainB.Codec.MustMarshal(txRaw)
+				data := types.IBCAccountPacketData{Type: types.EXECUTE_TX,
+					Data: txBytes}
+				packetData = data.GetBytes()
+			}, false,
+		},
+		{
+			"Invalid packet type", func() {
+				txBytes = []byte{}
+				// Type here is an ENUM
+				// Valid type is types.EXECUTE_TX
```

Comment:
Is there a zero value enum for `IBCAccountPacketData.Type` that is `UNKNOWN` or `UNSPECIFIED`? 
Protobuf normally recommends having the zero value enum as this for no-op or error scenarios and the enum types that you consume to invoke different code paths would come after.

For example:

```proto
enum Type {
    TYPE_UNSPECIFIED = 0;
    TYPE_EXECUTE_TX = 1;
}
```

Reply:
Updated this. Good catch & good to know. https://github.com/cosmos/ibc-go/pull/412/files#diff-5b066a6992f0f8e592492085bc4164eb73f17f8278ef7eafc064c5c197d80673R24

Reply:
Excellent! 🤝